azext_edge/edge/providers/check/mq.py

```diff
@@ -774,701 +476,243 @@ def evaluate_brokers(
-            route_type = next(iter(route))
-            route_details = route[route_type]
-
-            # shared properties
-            name = route_details.get("name")
-            check_manager.add_display(
-                target_name=target,
+            check_manager.add_target_eval(
+                target_name=target_listener_service,
                 namespace=namespace,
-                display=Padding(
-                    f"- Route: {{[bright_blue]{name}[/bright_blue]}}",
-                    padding,
-                ),
+                status=listener_service_eval_status,
+                value={"spec.clusterIP": cluster_ip},
             )
+        elif listener_spec_service_type.lower() == "nodeport":
+            pass
 
-            # route details are verbose
-            # TODO - table output?
-            if detail_level == ResourceOutputDetailLevel.verbose.value:
-                route_detail_padding = (0, 0, 0, padding[3] + PADDING_SIZE)
 
-                kafkaTopic = route_details.get("kafkaTopic")
-                mqttTopic = route_details.get("mqttTopic")
-                qos = route_details.get("qos")
+def _evaluate_broker_diagnostics_service(
+    check_manager: CheckManager,
+    target_brokers: str,
+    namespace: str,
+    detail_level: int = ResourceOutputDetailLevel.summary.value,
+) -> None:
+    diagnostics_service = get_namespaced_service(name=AIO_MQ_DIAGNOSTICS_SERVICE, namespace=namespace, as_dict=True)
+    if not diagnostics_service:
+        check_manager.add_target_eval(
+            target_name=target_brokers,
+            namespace=namespace,
+            status=CheckTaskStatus.error.value,
+            value=f"service/{AIO_MQ_DIAGNOSTICS_SERVICE} not found in namespace {namespace}",
+            resource_name=f"service/{AIO_MQ_DIAGNOSTICS_SERVICE}",
+        )
+        diag_service_desc_suffix = "[red]not detected[/red]."
+        diag_service_desc = (
+            f"Service {{[bright_blue]{AIO_MQ_DIAGNOSTICS_SERVICE}[/bright_blue]}} {diag_service_desc_suffix}"
+        )
+        check_manager.add_display(
+            target_name=target_brokers,
+            namespace=namespace,
+            display=Padding(
+                diag_service_desc,
+                (0, 0, 0, 12),
+            ),
+        )
+    else:
+        clusterIP = diagnostics_service.get("spec", {}).get("clusterIP")
+        ports: List[dict] = diagnostics_service.get("spec", {}).get("ports", [])
 
+        check_manager.add_target_eval(
+            target_name=target_brokers,
+            namespace=namespace,
+            status=CheckTaskStatus.success.value,
+            value={"spec": {"clusterIP": clusterIP, "ports": ports}},
+            resource_name=f"service/{AIO_MQ_DIAGNOSTICS_SERVICE}",
+        )
+        diag_service_desc_suffix = "[green]detected[/green]."
+        diag_service_desc = (
+            f"\nService {{[bright_blue]{AIO_MQ_DIAGNOSTICS_SERVICE}[/bright_blue]}} {diag_service_desc_suffix}"
```

Comment:
Sorry, meant that comment for this line, but same idea 😛

Reply:
got it will update!

Reply:
updated
<img width="251" alt="image" src="https://github.com/user-attachments/assets/30b8c6cf-e34d-40b5-aee5-d2c3b0166006">